src/Apps/Order/Components/CheckoutAddress.tsx

```diff
@@ -48,19 +72,126 @@ const VALIDATION_SCHEMA = Yup.object().shape({
   }),
 })
 
-const getDefaultCountry = (userCountry: string): string => {
-  const countryCode = ALL_COUNTRY_SELECT_OPTIONS.find(
-    country => country.text === userCountry
-  )?.value
-  return countryCode || "US"
-}
-
 export const CheckoutAddress: FC<{
   userCountry: string
   onChange: AddressChangeHandler
 }> = ({ userCountry, onChange }) => {
-  const { submitMutation: submitAddAddress } = useAddAddress()
-  const userDefaultCountry = getDefaultCountry(userCountry)
+  const userDefaultCountry = getDefaultCountry(userCountry, true)
+  const [displayModal, setDisplayModal] = useState(false)
+  const [modalTitle, setModalTitle] = useState<ErrorModalTitle>(
+    ErrorModalTitle.general
+  )
+  const [modalContent, setModalContent] = useState<ReactElement>()
+  const [selectedAddress, setSelectedAddress] = useState<
+    AddressSuggestionRadioButton
+  >(AddressSuggestionRadioButton.recommended)
+
+  const handleKeepAddressClick = () => {}
+
+  const handleEditAddressClick = () => {
+    setDisplayModal(false)
+  }
+
+  const handleUseSuggestedAddressClick = () => {}
+
+  const handleFormSubmit = async (address: Address) => {
+    const { firstLine, secondLine } = formatAddress(address)
+
+    const res = await mockRequest()
+    switch (res) {
+      case 0:
+        setModalTitle(ErrorModalTitle.general)
+        setModalContent(
```

Comment:
This is really unconventional: you should just change some state and let the component handle the rendering, rather than pushing JSX into state.

Reply:
thank you, I'll update.